### PR TITLE
[Python] Pyarrow integer filter pushdown bug fix

### LINKED ISSUE #9144: Error mapping from python pyarrow/numpy uint64 to duckdb LogicalType
### What happens?

If I have a pandas dataframe with a `uint64` column, I am able to use a `uint64` in a prepared statement. However, if that same dataframe is a pyarrow table, then it stops working.

### To Reproduce

```python
import pandas as pd
import pyarrow as pa
import numpy as np
import duckdb

val = np.iinfo(np.uint64).max
df = pd.DataFrame({"a": np.array([val], dtype=np.uint64)})

try:
    with duckdb.connect(database=":memory:", read_only=False) as con:
        _ = con.execute("select * from df where a = $1", (val,)).df()
except Exception as e:
    print("Attempt 1:", e)
else:
    print("Attempt 1: Success!")

t = pa.Table.from_pandas(df)
try:
    with duckdb.connect(database=":memory:", read_only=False) as con:
        _ = con.execute("select * from t where a = $1", (val,)).df()
except Exception as e:
    print("Attempt 2:", e)
else:
    print("Attempt 2: Success!")

try:
    with duckdb.connect(database=":memory:", read_only=False) as con:
        _ = con.execute("select * from t where a = $1", pa.array([val], type=pa.uint64())).df()
except Exception as e:
    print("Attempt 3:", e)
else:
    print("Attempt 3: Success!")

try:
    with duckdb.connect(database=":memory:", read_only=False) as con:
        _ = con.execute("select * from t where a = $1", np.array([val], dtype=np.uint64)).df()
except Exception as e:
    print("Attempt 4:", e)
else:
    print("Attempt 4: Success!")
```

gives

```
Attempt 1: Success!
Attempt 2: Invalid Error: OverflowError: Python int too large to convert to C long

At:
  pyarrow/error.pxi(144): pyarrow.lib.pyarrow_internal_check_status

Attempt 3: Not implemented Error: Unable to transform python value of type '<class 'pyarrow.lib.UInt64Scalar'>' to DuckDB LogicalType
Attempt 4: Not implemented Error: Unable to transform python value of type '<class 'numpy.uint64'>' to DuckDB LogicalType
```

### OS:

docker -- ubuntu 22.04

### DuckDB Version:

python-duckdb 0.8.1 from conda-forge

### DuckDB Client:

Python

### Full Name:

Brett Tully

### Affiliation:

Nearmap

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #9144

We called `pyarrow.dataset.scalar` with a value without explicitly setting the type, this caused pyarrow to create a scalar of a smaller type, and that somehow broke the arrow array stream scan.
Now we explicitly set the type

Also fixed up the `test_filter_pushdown.py` test, it was pretty much duplicated into `pyarrow_filter_pushdown.py` so I merged them, and added an extra test for this issue.

I also found an issue in an existing test in the file, https://github.com/duckdb/duckdb/pull/4817/files#r1339973721
I'd like to fix the test, I'm a little confused as to what the intent was, so for now I've skipped it instead

### CODE REVIEW

File: tools/pythonpkg/tests/fast/arrow/test_filter_pushdown.py [hunk 1]
```diff
@@ -137,314 +167,465 @@ def test_filter_pushdown_numeric(self, duckdb_cursor):
...
+            f"""
+            CREATE TABLE tbl as select {value}::{data_type} as i
+        """
+        )
+        expected = duckdb_cursor.table('tbl').fetchall()
+        filter = "i > 0"
+        rel = duckdb_cursor.table('tbl')
+        arrow_table = create_table(rel)
+        actual = duckdb_cursor.sql(f"select * from arrow_table where {filter}").fetchall()
+        assert expected == actual
```
Comment (bretttully) [inline tools/pythonpkg/tests/fast/arrow/test_filter_pushdown.py:495]: Does this make sense? And should we expect it to work?

```suggestion
        actual = duckdb_cursor.sql(f"select * from arrow_table where {filter}").fetchall()
        assert expected == actual

        # Test with equivalent prepared statement
        actual = duckdb_cursor.execute("select * from arrow_table where i > ?", (0,)).fetchall()
        assert expected == actual

        # Test equality
        actual = duckdb_cursor.execute("select * from arrow_table where i = ?", (value,)).fetchall()
        assert expected == actual
```

Comment (Tishj) [reply tools/pythonpkg/tests/fast/arrow/test_filter_pushdown.py:495]: Added it

**Comment by Mytherin** [comment]
Thanks!

**Comment by Tishj** [comment]
This skips a test because it was written incorrectly

The test should be fixed and no longer skipped
I left it as draft while waiting on a response to fix the test

File: tools/pythonpkg/tests/fast/arrow/test_filter_pushdown.py [hunk 2]
```diff
@@ -453,28 +641,39 @@ def test_filter_pushdown_2145(self, duckdb_cursor, pandas):
...
 
-        os.remove("data1.parquet")
-        os.remove("data2.parquet")
-
-    def test_filter_column_removal(self, duckdb_cursor):
-        duckdb_conn.execute("CREATE TABLE test AS SELECT range i, range j FROM range(5)")
-        duck_test_table = duckdb_conn.table("test")
-        arrow_test_table = duck_test_table.arrow()
-        duckdb_conn.register("arrow_test_table", arrow_test_table)
+    # https://github.com/duckdb/duckdb/pull/4817/files#r1339973721
```
Comment (Tishj) [inline tools/pythonpkg/tests/fast/arrow/test_filter_pushdown.py:656]: Note to self: This test should be fixed

**Pull request merged**